subprojects/composite-builds/src/integTest/groovy/org/gradle/integtests/composite/CompositeBuildContinueOnMultipleFailuresIntegrationTest.groovy

```diff
@@ -0,0 +1,156 @@
+
+    def setup() {
+        buildB = singleProjectBuild('buildB') {
+            buildFile << javaProject()
+            file('src/test/java/SampleTestB.java') << junitTestClass('SampleTestB')
+        }
+
+        buildC = multiProjectBuild('buildC', ['sub1', 'sub2', 'sub3']) {
+            buildFile << """
+                allprojects {
+                    ${javaProject()}
+                }
+
+                test.dependsOn 'sub1:test', 'sub2:test', 'sub3:test'
+            """
+            file('sub1/src/test/java/SampleTestC_Sub1.java') << junitTestClass('SampleTestC_Sub1')
+            file('sub2/src/test/java/SampleTestC_Sub2.java') << junitTestClass('SampleTestC_Sub2')
+            file('sub3/src/test/java/SampleTestC_Sub3.java') << junitTestClass('SampleTestC_Sub3')
+        }
+
+        buildD = singleProjectBuild('buildD') {
+            buildFile << javaProject()
+            file('src/test/java/SampleTestD.java') << junitTestClass('SampleTestD')
+        }
+    }
+
+    def "can collect build failures from multiple included builds"() {
+        when:
+        includedBuilds << buildB << buildC << buildD
+
+        buildA.buildFile << """
+            task testAll {
+                dependsOn gradle.includedBuilds*.task(':test')
+            }
+        """
+
+        and:
+        fails(buildA, 'testAll', [CONTINUE_COMMAND_LINE_OPTION])
+
+        then:
+        !errorOutput.contains('Multiple build failures')
+        assertTaskExecuted(':buildB', ':test')
+        assertTaskExecuted(':buildC', ':sub1:test')
+        assertTaskExecuted(':buildC', ':sub2:test')
+        assertTaskExecuted(':buildC', ':sub3:test')
+        assertTaskExecuted(':buildD', ':test')
+        assertTaskExecutionFailureMessage(errorOutput, 1, ':buildB:test')
+        assertTaskExecutionFailureMessage(errorOutput, 2, ':buildC:sub1:test')
+        assertTaskExecutionFailureMessage(errorOutput, 3, ':buildC:sub2:test')
+        assertTaskExecutionFailureMessage(errorOutput, 4, ':buildC:sub3:test')
+        assertTaskExecutionFailureMessage(errorOutput, 5, ':buildD:test')
+    }
+
+    def "can collect build failure in root and included build"() {
+        when:
+        includedBuilds << buildC
+
+        buildA.buildFile << """
+            ${mavenCentralRepository()}
+            ${junitDependency()}
+
+            task testAll {
+                dependsOn 'test'
+                dependsOn gradle.includedBuilds*.task(':test')
+            }
+        """
+        file('buildA/src/test/java/SampleTestA.java') << junitTestClass('SampleTestA')
+
+        and:
+        fails(buildA, 'testAll', [CONTINUE_COMMAND_LINE_OPTION])
+
+        then:
+        !errorOutput.contains('Multiple build failures')
+        assertTaskExecuted(':', ':test')
+        assertTaskExecuted(':buildC', ':sub1:test')
+        assertTaskExecuted(':buildC', ':sub2:test')
+        assertTaskExecuted(':buildC', ':sub3:test')
+        assertTaskExecutionFailureMessage(errorOutput, 1, ':test')
```

Comment:
We should probably be using `failure.assertHasDescription()` and `failure.assertHasCause()` here.
If the infrastructure doesn't support multiple exceptions then this would be a good time to add support.

I like the fact that we are verifying the exception number. I think adding some support for multiple failures to `org.gradle.integtests.fixtures.executer.ExecutionFailure` would be a good thing.

Reply:
I will address this as a separate PR. That'll allow us to get the change into the 4.6 release. It also makes the review of the second PR easier.